core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -320,6 +320,31 @@ struct math_function_name;
   };                                                                  \
   constexpr char math_function_name<MathUnaryFunction_##FUNC>::name[]
 
+#define DEFINE_UNARY_FUNCTION_EVAL_CUSTOM(FUNC, ULP_FACTOR, REF_FUNC) \
+  struct MathUnaryFunction_##FUNC {                                   \
+    template <typename T>                                             \
+    static KOKKOS_FUNCTION auto eval(T x) {                           \
+      static_assert(                                                  \
+          std::is_same<decltype(Kokkos::FUNC((T)0)),                  \
+                       math_unary_function_return_type_t<T>>::value); \
+      return Kokkos::FUNC(x);                                         \
+    }                                                                 \
+    template <typename T>                                             \
+    static auto eval_std(T x) {                                       \
```

Comment:
is never called

Reply:
It is called via https://github.com/kokkos/kokkos/blob/90bf117bc9890453018f62853cc15473f7d3324e/core/unit_test/TestMathematicalFunctions.hpp#L509

Reply:
Regarding the macro and function naming please also see my previous comment, https://github.com/kokkos/kokkos/pull/5644#issuecomment-1330928711